### DISCUSSION TITLE
ScriptReference/Editor.OnSceneGUI

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#24] [Notes]
# ScriptReference/Editor.OnSceneGUI

https://docs.unity3d.com/ScriptReference/Editor.OnSceneGUI.html

**Comment by vertxxyz** [comment]
Drawing IMGUI content in OnSceneGUI can be done within a [`Handles.BeginGUI()`](https://docs.unity3d.com/ScriptReference/Handles.BeginGUI.html) and [`Handles.EndGUI()`](https://docs.unity3d.com/ScriptReference/Handles.EndGUI.html) scope.

**Comment by vertxxyz** [comment]
Editors for ScriptableObjects do not call this method. Use [`SceneView.duringSceneGui`](https://docs.unity3d.com/ScriptReference/SceneView-duringSceneGui.html) instead.

**Comment by vertxxyz** [comment]
OnSceneGUI is called once per `target`. So making a multi-object inspector does not require using [`targets`](https://docs.unity3d.com/ScriptReference/Editor-targets.html).
